handsontable/test/e2e/focus/focusHandling.spec.js

```diff
@@ -0,0 +1,182 @@
+        hot.getSelectedRangeLast().highlight.row,
+        hot.getSelectedRangeLast().highlight.col,
+        true
+      ));
+    });
+  });
+
+  describe('`imeFastEdit` enabled', () => {
+    it('should refocus the browser on the active editor\'s `TEXTAREA` element after a delay specified in the focus' +
+      ' manager', async() => {
+      const hot = handsontable({
+        data: Handsontable.helper.createSpreadsheetData(10, 10),
+        imeFastEdit: true,
+      });
+
+      expect(document.activeElement).toEqual(document.body);
+
+      hot.selectCell(2, 2);
+
+      expect(document.activeElement).toEqual(hot.getCell(
+        hot.getSelectedRangeLast().highlight.row,
+        hot.getSelectedRangeLast().highlight.col,
+        true
+      ));
+
+      await sleep(hot.getFocusManager().getRefocusDelay());
+
+      expect(document.activeElement).toEqual(hot.getActiveEditor().TEXTAREA);
+
+      hot.selectCells([[1, 1, 3, 3], [4, 4, 2, 2]]);
+
+      expect(document.activeElement).toEqual(hot.getCell(
+        hot.getSelectedRangeLast().highlight.row,
+        hot.getSelectedRangeLast().highlight.col,
+        true
+      ));
+
+      await sleep(hot.getFocusManager().getRefocusDelay());
+
+      expect(document.activeElement).toEqual(hot.getActiveEditor().TEXTAREA);
+    });
+
+    it('should not refocus the browser if the active editor doesn\'t contain a `TEXTAREA` element', async() => {
+      const hot = handsontable({
+        data: Handsontable.helper.createSpreadsheetData(10, 10),
+        imeFastEdit: true,
+        type: 'checkbox'
+      });
+
+      expect(document.activeElement).toEqual(document.body);
+
+      hot.selectCell(2, 2);
+
+      expect(document.activeElement).toEqual(hot.getCell(
+        hot.getSelectedRangeLast().highlight.row,
+        hot.getSelectedRangeLast().highlight.col,
+        true
+      ));
+
+      await sleep(hot.getFocusManager().getRefocusDelay());
+
+      expect(document.activeElement).toEqual(hot.getCell(
+        hot.getSelectedRangeLast().highlight.row,
+        hot.getSelectedRangeLast().highlight.col,
+        true
+      ));
+    });
+
+    it('should be possible to force refocusing on a different element than active editor\'s `TEXTAREA` by providing' +
+      ' a refocus element getter', async() => {
+
+      class CustomEditor extends Handsontable.editors.TextEditor {
+        createElements() {
+          super.createElements();
+
+          this.TEXTAREA_ALTERNATIVE = this.hot.rootDocument.createElement('input');
+          this.TEXTAREA_PARENT.appendChild(this.TEXTAREA_ALTERNATIVE);
+        }
+      }
+
+      const hot = handsontable({
+        data: Handsontable.helper.createSpreadsheetData(10, 10),
+        imeFastEdit: true,
+        editor: CustomEditor
+      });
+
+      expect(document.activeElement).toEqual(document.body);
+
+      hot.getFocusManager().setRefocusElementGetter(() => hot.getActiveEditor().TEXTAREA_ALTERNATIVE);
```

Comment:
Just suggestion. Instead, require a developer to use another API for focusing the custom editor element (`getFocusManager()`). Maybe it would be better that the focus manager would focus element by [`data-hot-input`](https://github.com/handsontable/handsontable/blob/develop/handsontable/src/editors/textEditor/textEditor.js#L220) attribute that has been required for some time by all editors including custom ones.